quilt/data.py

```diff
@@ -56,38 +54,43 @@ def __getattr__(self, name):
         path = self._prefix + '/' + name
 
         try:
-            node = self._package.get(path)
+            return DataNode(self._package, path)
         except PackageException:
             raise AttributeError("No such table or group: %s" % path)
 
-        if isinstance(node, GroupNode):
-            return DataNode(self._package, path)
-        else:
-            return self._package.get_obj(node)
-
     def __repr__(self):
         cinfo = str(self.__class__)
         finfo = 'File: ' + self._package.get_path()
-        pinfo = 'Path: ' + self._prefix + '/'
+        pinfo = 'Path: ' + self._prefix + ('/' if self._is_group() else '')
         #TODO maybe show all descendant subpaths instead of just children
-        groups = sorted(k + '/' for k in self._groups())
-        dfs = sorted(self._dfs())
+        groups = sorted(k + '/' for k in self._group_keys())
+        dfs = sorted(self._df_keys())
         output = [cinfo, finfo, pinfo] + groups + dfs
         return '\n'.join(output)
 
     def __dir__(self):
         # https://mail.python.org/pipermail/python-ideas/2011-May/010321.html
         return sorted(set((dir(type(self)) + list(self.__dict__) + self._keys())))
 
-    def _dfs(self):
+    def _is_group(self):
+        return isinstance(self._node, GroupNode)
+
+    def _is_df(self):
```

Comment:
We shouldn't report a node as a df unless it's actually a meaningful Pandas DF. Files, e.g., should not be dfs. Do we need a new concept, e.g., LeafNode that specifies merely that a node has no children?